tests/v2/validation/rbac/clusterandprojectroles/cluster_role_test.go

```diff
@@ -0,0 +1,149 @@
+	require.NoError(rb.T(), err)
+
+	rb.T().Logf("Adding user as " + rbac.ClusterOwner.String() + " to the downstream cluster.")
+	err = users.AddClusterRoleToUser(rb.client, rb.cluster, clusterAdmin, rbac.ClusterOwner.String(), nil)
+	require.NoError(rb.T(), err)
+	clusterAdminClient, err = clusterAdminClient.ReLogin()
+	require.NoError(rb.T(), err)
+
+	clusterOwnerProject, err := clusterAdminClient.Management.Project.Create(projects.NewProjectConfig(rb.cluster.ID))
+	require.NoError(rb.T(), err)
+	err = users.AddProjectMember(clusterAdminClient, clusterOwnerProject, additionalUser, rbac.ProjectOwner.String(), nil)
+	require.NoError(rb.T(), err)
+
+	projectListClusterAdditionalUser, err := projects.ListProjectNames(additionalUserClient, rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	assert.Equal(rb.T(), 1, len(projectListClusterAdditionalUser))
+	assert.Equal(rb.T(), clusterOwnerProject.Name, projectListClusterAdditionalUser[0])
+
+	err = users.RemoveProjectMember(clusterAdminClient, additionalUser)
+	require.NoError(rb.T(), err)
+	projectListClusterAdditionalUser, err = projects.ListProjectNames(additionalUserClient, rb.cluster.ID)
+	require.Empty(rb.T(), projectListClusterAdditionalUser)
+}
+
+func (rb *RBACAdditionalTestSuite) TestClusterOwnerAddsUserAsClusterOwner() {
+	subSession := rb.session.NewSession()
+	defer subSession.Cleanup()
+
+	clusterAdmin, clusterAdminClient, err := rbac.SetupUser(rb.client, rbac.StandardUser.String())
+	require.NoError(rb.T(), err)
+
+	errUserRole := users.AddClusterRoleToUser(rb.client, rb.cluster, clusterAdmin, rbac.ClusterOwner.String(), nil)
+	require.NoError(rb.T(), errUserRole)
+	clusterAdminClient, err = clusterAdminClient.ReLogin()
+	require.NoError(rb.T(), err)
+
+	additionalUser, additionalUserClient, err := rbac.SetupUser(rb.client, rbac.StandardUser.String())
+	require.NoError(rb.T(), err)
+
+	errUserRole2 := users.AddClusterRoleToUser(clusterAdminClient, rb.cluster, additionalUser, rbac.ClusterOwner.String(), nil)
+	require.NoError(rb.T(), errUserRole2)
+	additionalUserClient, err = additionalUserClient.ReLogin()
+	require.NoError(rb.T(), err)
+
+	clusterList, err := additionalUserClient.Steve.SteveType(clusters.ProvisioningSteveResourceType).ListAll(nil)
+	require.NoError(rb.T(), err)
+	assert.Equal(rb.T(), 1, len(clusterList.Data))
+
+	err = users.RemoveClusterRoleFromUser(clusterAdminClient, additionalUser)
+	require.NoError(rb.T(), err)
+	clusterList, err = additionalUserClient.Steve.SteveType(clusters.ProvisioningSteveResourceType).ListAll(nil)
+	require.Empty(rb.T(), clusterList.Data)
+}
+
+func (rb *RBACAdditionalTestSuite) TestClusterOwnerAddsClusterMemberAsProjectOwner() {
+	subSession := rb.session.NewSession()
+	defer subSession.Cleanup()
+
+	clusterAdmin, clusterAdminClient, err := rbac.SetupUser(rb.client, rbac.StandardUser.String())
+	require.NoError(rb.T(), err)
+
+	errUserRole := users.AddClusterRoleToUser(rb.client, rb.cluster, clusterAdmin, rbac.ClusterOwner.String(), nil)
```

Comment:
what are 'errUserRole's? Since this isn't reflected in the test name, its a little confusing.

Reply:
Yeah the name is confusing. Used to errors just being named `err`.